_posts/blog/2017-08-014-upday_architecture.md

```diff
@@ -0,0 +1,106 @@
+
+### Where to draw the technologies diversity line
+
+Once living in this wonderful and interesting microservices world, **it was nice to have the opportunity to use new technologies** without being restricted to just the original ones. 
+However, not everything is pink and full of rainbows in the microservices world. **You have to know when to stop** creating new microservices or adding more and more new technologies to your stack. 
+Because your team - whose members could always change - has to continue maintaining and understanding this zoo of services and its underlying technologies. 
+
+Like everything in life: **You have to find a balance.** Since everyone has different opinions, there is no one perfect solution. 
+Also, something that is *en vogue* now may not be the *silver bullet* forever. 
+It might be that the best approach for your team is having less services and only one single technology. 
+Or it may be that applying different technologies and trying new ones is what the developers want to do, because they are hungry to learn and face new challenges. 
+**Whatever the situation is, just don’t lose the focus on what your product really requires.**
+
+### Our architecture today
+
+As a team of young developers who strive for challenges, we have evolved into our current setup where we have better small services.
+**But we never create a new microservice for the sake of creating a microservice**, only when there are reasons to go for a new one.
+And this seems to work quite well for both, our developers and our product.
+
+Our *Android* app we serve has two main parts: 
+
+- the _Top News section_ (news everyone needs to know about) and the 
+- _My News section_ (based on the user’s interests and their interactions with our app). 
+
+We decided to keep these areas separated and independent, **basing their communication only on messaging**, as described in the **SCS** [*(Self Contained Systems)*](http://scs-architecture.org/) approach. 
+So when there is any problem related with *Top News*, *My News* is not affected and the other way around. 
+
+This is how our architecture looks like from a bird’s eyes view (very high level overview):
+
+<img style="margin: auto; margin-left: 15%; margin-top: 10px;" src="/images/blog/upday_architecture/high-level_arch_overview.jpg"/><br/>
+
+And here a more detailed overview of the (micro-) services - not always so *"super micro"* - developed with different technologies, composing the **Top News System**:
+
+<img style="margin: auto; margin-left: 2%; margin-top: 10px;" src="/images/blog/upday_architecture/microservices_top_news_system.jpg"/><br/>
+
+### Our learnings and conclusions so far
+
+*… which are not an universal truth and may not even for us last forever:*
+
+- Although we have mastered *Java 8* and *Spring Boot* technologies better so far, **we might move to other technologies if they fit our requirements or problems better.** But this learning has a big BUT: 
```

Comment:
Although we have mastered *Java 8* and *Spring Boot* technologies so far, **we might move to other technologies if they fit our requirements or problems better.** But this learning has a big BUT: